control_panel_api/serializers.py

```diff
@@ -266,3 +269,41 @@ class Meta:
             'nickname',
             'name',
         )
+
+
+class ESBucketHitsSerializer(serializers.BaseSerializer):
+
+    def to_representation(self, aggregations):
+        accessed_by_sum_map = {}
+        accessed_by_type_map = {}
+
+        for result in aggregations.bucket_hits:
+            role_type, accessed_by = self._extract_fields(result.key)
+
+            accessed_by_type_map[accessed_by] = role_type
+
+            if accessed_by not in accessed_by_sum_map:
+                accessed_by_sum_map[accessed_by] = result.doc_count
+            else:
+                accessed_by_sum_map[accessed_by] += result.doc_count
```

Comment:
If you use `.setdefault()` or `defaultdict(int)` you can avoid this test, eg:
```python
accessed_by_sum_map = collections.defaultdict(int)
for result in ...:
    ...
    accessed_by_sum_map[accessed_by] += result.doc_count
```

Reply:
[Ruby's hashes](https://ruby-doc.org/core-2.5.0/Hash.html) has something similar which is handy in these cases and I was wondering if you could do the same here.

Nice tip 👍

Reply:
(Troll mode on: In Ruby you would just do `accessed_by_sum_map = Hash(0)` because it's a beautiful language :trollface:  but it's the same thing)